include/SFML/System/Angle.hpp

```diff
@@ -0,0 +1,458 @@
+    // Static member data
+    ////////////////////////////////////////////////////////////
+    SFML_SYSTEM_API static const Angle Zero; //!< Predefined 0 degree angle value
+
+private:
+
+    friend constexpr Angle degrees(float);
+    friend constexpr Angle radians(float);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Construct from a number of radians
+    ///
+    /// This function is internal. To construct angle values,
+    /// use sf::radians or sf::degrees instead.
+    ///
+    /// \param radians Angle in radians
+    ///
+    ////////////////////////////////////////////////////////////
+    constexpr explicit Angle(float radians);
+
+private:
+
+    ////////////////////////////////////////////////////////////
+    // Member data
+    ////////////////////////////////////////////////////////////
+    float m_radians; //!< Angle value stored as radians
+};
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Construct an angle value from a number of degrees
+///
+/// \param amount Number of degrees
+///
+/// \return Angle value constructed from the amount of degrees
+///
+/// \see radians
+///
+////////////////////////////////////////////////////////////
+constexpr Angle degrees(float angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Construct an angle value from a number of radians
+///
+/// \param amount Number of radians
+///
+/// \return Angle value constructed from the amount of radians
+///
+/// \see degrees
+///
+////////////////////////////////////////////////////////////
+constexpr Angle radians(float angle);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of == operator to compare two angle values
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return True if both angle values are equal
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr bool operator ==(Angle left, Angle right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of != operator to compare two angle values
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return True if both angle values are different
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr bool operator !=(Angle left, Angle right);
+
+////////////////////////////////////////////////////////////
+/// \relates Angle
+/// \brief Overload of < operator to compare two angle values
+///
+/// \param left  Left operand (an angle)
+/// \param right Right operand (an angle)
+///
+/// \return True if \a left is lesser than \a right
+///
+////////////////////////////////////////////////////////////
+[[nodiscard]] constexpr bool operator <(Angle left, Angle right);
```

Comment:
Because these operators don't modify any inputs and modify no global state, the only observable behavior is in the form of its return value so it must be used. This is the case with `sf::Time`s comparison operators as well.